File: Virucodesoup/Ruby/External/FXRuby/Text Example/textExample.rb
# ===========================
# File : textExample.rb
#
# Credit:
# - Coded by (aka in cyber) Vsio Stitched
#
# Misc:
# - Written in Ruby programming language
#
# License:
# - Free to use
# - May include me or not in credit if included in other project
# ===========================

require 'fox16'
include Fox

class TextExample < FXMainWindow
	def initialize(_app, _title, _w, _h)
	# inits value when instantiated
		super(_app, _title, :width => _w, :height => _h)  
	end
	
	def create()
	# to activate create event from parent class
		super
		show(PLACEMENT_SCREEN)
	end
	
	def showText()
	# shows text
		FXLabel.new(self,"This is an example of displaying text. You can use FXLabel to achieve it.")
	end
	
	def driver()
	# performs object functionally
		showText()
	end
end

File: Virucodesoup/Javascript/External/Crafty/Text Example/textExample.js
// ===========================
// File : textExample.js
//
// Credit:
// - Coded by (aka in cyber) Vsio Stitched
//
// Misc:
// - Written in Javascript programming language
// - Included Extension: Crafty
// 
// License:
// - Free to use
// - May include me or not in credit if included in other project
// ===========================

Crafty.c("TextExample", {
	init: function() {
		
		// INITIALIZE
		
		this.addComponent("Text","2D","DOM");
		
		
		// METHOD
		
		this.drawText = function(_text,_w,_h,_newX,_newY) 
		// draws Text
		{
			this.attr({w:_w,h:_h,x:_newX,y:_newY});
			this.text(_text).css({"text-align": "center","vertical-align": "middle"});;
		}
	}	
});

function TextExample_Driver() 
// performs object's functionality
{
	var te = Crafty.e("TextExample");
	te.drawText("This is text example",480,100,0,100);
}
